flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/StopWithSavepoint.java

```diff
@@ -178,27 +168,6 @@ private void completeOperationAndGoToFinished(String savepoint) {
         context.goToFinished(ArchivedExecutionGraph.createFrom(getExecutionGraph()));
     }
 
-    private void handleAnyFailure(Throwable cause) {
-        operationFailureCause = cause;
```

Comment:
Are you sure this is safe? I don't know if there's a guarantee that the savepoint future is completed (exceptionally) before the state is notified of a task failure.